interfaces/builtin/kubernetes_support.go

```diff
@@ -174,20 +174,12 @@ var kubernetesSupportConnectedPlugKmodKubeProxy = []string{
 	`stp`,
 }
 
-type kubernetesSupportInterface struct{}
-
-func (iface *kubernetesSupportInterface) Name() string {
-	return "kubernetes-support"
+type kubernetesSupportInterface struct {
+	commonInterface
 }
 
-func (iface *kubernetesSupportInterface) StaticInfo() interfaces.StaticInfo {
-	return interfaces.StaticInfo{
-		Summary:              kubernetesSupportSummary,
-		BaseDeclarationPlugs: kubernetesSupportBaseDeclarationPlugs,
-		BaseDeclarationSlots: kubernetesSupportBaseDeclarationSlots,
-		ImplicitOnCore:       true,
-		ImplicitOnClassic:    true,
-	}
+func (iface *kubernetesSupportInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
+	return sanitizeSlotReservedForOS(iface, slot)
```

Comment:
You may want to run "super" version of the method as well:`iface.commonInterface.BeforePrepareSlot` though I down't think we do much validation there now.